src/sys/unix/pipe.rs

```diff
@@ -172,9 +182,7 @@ pub fn new() -> io::Result<(Sender, Receiver)> {
         }
 
         for fd in &fds {
-            if libc::fcntl(*fd, libc::F_SETFL, libc::O_NONBLOCK) != 0
-                || libc::fcntl(*fd, libc::F_SETFD, libc::FD_CLOEXEC) != 0
-            {
+            if libc::fcntl(*fd, libc::F_SETFL, libc::O_NONBLOCK | libc::FD_CLOEXEC) != 0 {
```

Comment:
As pointed out by @Thomasdezeeuw elsewhere: "This uses `F_SETFD` and `F_SETFL` (`FD` vs `FL`), two different commands. So this doesn't work."